state/accounts/esdtAsBalance.go

```diff
@@ -107,6 +103,15 @@ func createEmptyESDT() *esdt.ESDigitalToken {
 	return &esdt.ESDigitalToken{Value: big.NewInt(0), Type: uint32(core.Fungible)}
 }
 
+func fillMandatoryFields(esdtData *esdt.ESDigitalToken) {
+	// make extra sure we have these fields set
+	if esdtData.Value == nil {
+		esdtData.Value = big.NewInt(0)
+	}
+
+	esdtData.Type = uint32(core.Fungible)
```

Comment:
This is not the right implementation to enforce the type here. Starting with `RunTypeComponents` the type should be validated based on `NativeESDT` from config. Getting the balance from account should not change token type, we just use add/sub an amount, the type should always be valid.

Reply:
We just load a token name from config, it has no properties.
Also, the native token should be and only makes sense to be `Fungible`